src/averageAges.js

```diff
@@ -37,7 +45,12 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter(woman => people.some(kid => kid.mother === woman.name))
+    : people.filter(woman => woman.sex === 'f');
+
+  return women.map(item => item.died - item.born)
+    .reduce((sum, age) => sum + age / women.length, 0);
```

Comment:
create a separate method to calculate an average age, you use it in different places